src/app/models/main.rs

```diff
@@ -7,7 +8,13 @@ use crate::app::components::utils::format_duration;
 
 impl From<&AlbumDescription> for AlbumModel {
     fn from(album: &AlbumDescription) -> Self {
-        AlbumModel::new(&album.artists_name(), &album.title, &album.art, &album.id)
+        AlbumModel::new(
+            &album.artists_name(),
+            &album.title,
+            album.year(),
```

Comment:
This doesn't compile because `AlbumDescription` does not have `year()` or `released_date`. Using `FullAlbum, AlbumFullDescription` would do?

Reply:
We could make it an Option I suppose! (see main answer to the issue)